src/components/FolderDetails.js

```diff
@@ -12,7 +15,25 @@ import '../components/FolderDetails.css';
 export default function FolderDetails({ folderListData }) {
   const [windowWidth, setWindowWidth] = useState(window.innerWidth);
   const [selectedFolder, setSelectedFolder] = useState('all');
-  const selectedFolderData = useFetchData('targetUserFolderLinkList', 1, selectedFolder) || [];
+  const [modals, setModals] = useState({
+    addModal: false,
+    editModal: false,
+    removeModal: false,
+    shareFolderModal: false,
+  });
```

Comment:
만약 동시에 여러 모달을 띄울 수 있도록 설계한다면 이런 식으로 작성할 수 있습니다.
하지만 한 번에 하나의 모달만 띄운다면, 이렇게 `각자의 모달 상태`를 관리할 필요 없이 `현재 켜진 모달이 뭔지`를 관리하시면 됩니다.

예를 들면,
```suggestion
  const [currentModal, setCurrentModal] = useState('');
```

이런 식으로 state를 작성하고,

모달 state에 매핑되는 실제 모달 컴포넌트는 다음처럼 객체로 관리할 수 있습니다.
```js
  const MODAL_MAP = {
    addModal : <EditAndAddModal modalTitle="폴더 추가" buttonText="추가하기" onClose={() => toggleModal('addModal')} />,
    ...
  }
```

그리고 실제로 출력하는 곳에서는
```js
  MODAL_MAP[currentModal]
```

이런 식으로 작성할 수 있겠네요.

그렇다면 닫는 로직은 이런 식으로 되겠죠?
```js
  const closeModal = () => {
    setCurrentModal('');
  }